editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java

```diff
@@ -0,0 +1,359 @@
+	}
+
+	@Override
+	public boolean performFinish() {
+		try {
+			generate();
+		} catch (Exception e) {
+			Wsdl2RestUIActivator.pluginLog().logError(e);
+			ErrorDialog.openError(
+					getShell(),
+					UIMessages.wsdl2RestWizardErrorWindowTitle,
+					UIMessages.wsdl2RestWizardErrorMessage,
+					new Status(IStatus.ERROR, Wsdl2RestUIActivator.PLUGIN_ID, e.getMessage(), e));
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Uses the Workbench selection service to find the currently selected project and use it as the source. 
+	 * @return IProject
+	 */
+	protected IProject getSelectedProjectFromSelectionService() {
+		ISelectionService ss = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getSelectionService();
+		String projExpID = "org.eclipse.ui.navigator.ProjectExplorer"; //$NON-NLS-1$
+		ISelection sel = ss.getSelection(projExpID);
+		Object selectedObject=sel;
+		if(sel instanceof IStructuredSelection) {
+			selectedObject=
+					((IStructuredSelection)sel).getFirstElement();
+		}
+		if (selectedObject instanceof IAdaptable) {
+			IResource res = ((IAdaptable) selectedObject).getAdapter(IResource.class);
+			return res.getProject();
+		}
+		return null;
+	}	
+
+	@Override
+	public void addPages() {
+		super.addPages();
+
+		// page one
+		IProject initialProject = getSelectedProjectFromSelectionService();
+		Wsdl2RestWizardFirstPage pageOne = new Wsdl2RestWizardFirstPage("page1", //$NON-NLS-1$ 
+				UIMessages.wsdl2RestWizardPageOneTitle);
+		if (initialProject != null) {
+			options.setProjectName(initialProject.getName());
+		}
+		addPage(pageOne);
+
+		pageTwo = new Wsdl2RestWizardSecondPage("page2", //$NON-NLS-1$ 
+				UIMessages.wsdl2RestWizardPageTwoTitle); 
+		addPage(pageTwo);
+	}
+
+	public boolean isProjectBlueprint(IProject project) {
+		CamelMavenUtils cmu = new CamelMavenUtils();
+		List<Dependency> projectDependencies = cmu.getDependencyList(project);
+		Iterator<Dependency> depIter = projectDependencies.iterator();
+		while(depIter.hasNext()) {
+			Dependency dependency = depIter.next();
+			if ("org.apache.camel".equals(dependency.getGroupId()) && //$NON-NLS-1$
+					"camel-blueprint".equals(dependency.getArtifactId())) { //$NON-NLS-1$
+				return true;
+			}
+		}
+		return false;
+	}
+	
+	public boolean isProjectSpringBoot(IProject project) {
+		CamelMavenUtils cmu = new CamelMavenUtils();
+		List<Dependency> projectDependencies = cmu.getDependencyList(project);
+		return CamelCatalogUtils.hasSpringBootDependency(projectDependencies);
+	}
+
+	private void updateDependencies() {
+		List<org.fusesource.ide.camel.model.service.core.catalog.Dependency> deps = new ArrayList<>();
+		org.fusesource.ide.camel.model.service.core.catalog.Dependency one = 
+				new org.fusesource.ide.camel.model.service.core.catalog.Dependency();
+		one.setArtifactId("jboss-jaxrs-api_2.0_spec"); //$NON-NLS-1$
+		one.setGroupId("org.jboss.spec.javax.ws.rs"); //$NON-NLS-1$
+		one.setVersion("1.0.0.Final-redhat-1"); //$NON-NLS-1$
```

Comment:
where this version comes from? How it will evolve? Should we plan in a following iteration to have it configurable from the online "configuration" files depending on the Camel versions? https://github.com/jbosstools/jbosstools-fuse/pull/1138#discussion_r180427333

Reply:
the method updateDependencies is reported as not used, so we can remove it and seems the version is useless :-)

Reply:
Version was lifted from one of the original examples I found and am unable to find the same version in any examples today. I'm open to suggestions on how to remedy it.

Reply:
ok as it is working with this version currently. I think that we can merge the first iteration as-is on this point.
I created an issue to investigate if we need to have a version that needs to be modifiable across camel versions and/or if we need to use a different one: https://issues.jboss.org/browse/FUSETOOLS-2971